src/lib/services/searchUtilActions/AasSearcher.ts

```diff
@@ -88,133 +83,105 @@ export class AasSearcher {
-                            aasData: null,
-                        };
-                    aas = potentiallyMultipleAas[0].aas;
-                }
-            }
+    async performFullSearch(searchInput: string): Promise<AasSearchResult> {
+        const aasIds = await this.performAasDiscoverySearch(searchInput);
+        const foundMultipleDiscoveryResults = aasIds && aasIds.length > 1;
+        const foundOneDiscoveryResult = aasIds && aasIds.length === 1;
 
-            const aasData =
-                registrySearchResult?.registryAasData != null
-                    ? {
-                          submodelDescriptors: registrySearchResult.registryAasData.submodelDescriptors,
-                          aasRegistryRepositoryOrigin: registrySearchResult.registryAasData.aasRegistryRepositoryOrigin,
-                      }
-                    : null;
+        if (foundMultipleDiscoveryResults) {
+            return this.createDiscoveryRedirectResult(searchInput);
+        }
 
-            // If not found: Error: AAS could not be found
+        const aasId = foundOneDiscoveryResult ? aasIds[0] : searchInput;
+        const aasIdEncoded = encodeBase64(aasId);
 
-            return {
-                redirectUrl: `/viewer/${encodeBase64(aas.id)}`,
-                aas: aas,
-                aasData: aasData,
-            };
+        const aasRegistryResult = await this.performRegistrySearch(aasId);
+        if (aasRegistryResult) {
+            return aasRegistryResult;
         }
+
+        const defaultResult = await this.multipleDataSource.getAasFromDefaultRepository(aasIdEncoded);
+        if (defaultResult) {
+            return this.createAasResult(defaultResult);
+        }
+
+        const potentiallyMultipleAas = await this.multipleDataSource.getAasFromAllRepos(aasIdEncoded);
+        if (potentiallyMultipleAas.length > 1) {
+            return this.createDiscoveryRedirectResult(searchInput);
+        }
+        return this.createAasResult(potentiallyMultipleAas[0].aas);
     }
 
-    /**
-     * Resolves the given Asset ID using the discovery service.
-     *
-     * This function takes an Asset ID and attempts to resolve it using a discovery service.
-     * If the Asset ID is found, it returns the resolved AAS ID.
-     * If not found, it returns `null`.
-     *
-     * @param {string} searchAssetId - The Asset ID to resolve using the discovery service.
-     * @returns {Promise<string | null>} A promise that resolves to the AAS ID as a string, or `null` if the Asset ID is not found.
-     */
-    async performAasDiscoverySearch(searchAssetId: string): Promise<string[] | null> {
-        try {
-            if (!searchAssetId) {
-                throw new NotFoundError();
-            }
-            const aasIds = (await this.discoveryServiceClient.getAasIdsByAssetId(searchAssetId)).result;
+    // TODO: handle multiple endpoints as result
+    public async performRegistrySearch(searchAasId: string): Promise<AasSearchResult | null> {
```

Comment:
This is a "performSearchAndFetchResult" combination method.
Maybe name it accordingly